test/parallel/test-http-agent-maxsockets.js

```diff
@@ -0,0 +1,55 @@
+var common = require('../common');
+var assert = require('assert');
+var http = require('http');
+var Agent = require('_http_agent').Agent;
+
+var agent = new Agent({
+  keepAlive: true,
+  keepAliveMsecs: 1000,
+  maxSockets: 2,
+  maxFreeSockets: 2
+});
+
+var server = http.createServer(function (req, res) {
```

Comment:
Style nit: can you be consistent with `function() {` vs. `function () {` in this file?  The former is the preferred way in io.js.